### ISSUE TITLE
compact BigFloat printing

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#23382]
Since https://github.com/JuliaLang/julia/pull/23334, all fixed-size floating-point types compact-print `NaN` whereas big-float prints NaNs like this:
```jl
julia> show(IOContext(STDOUT, :compact => true), big(NaN))
BigFloat(NaN, 256)
```
However, this is part of a somewhat broader issue, which is that BigFloats don't really do compact printing:
```jl
julia> show(IOContext(STDOUT, :compact => true), big(pi))
3.141592653589793238462643383279502884197169399375105820974944592307816406286198
```
We should consider shortening the compact printing of BigFloats.

**Comment by jw3126** [comment]
About the second point, the `showcompact` function does not omit elements from huge arrays either. So it would be inconsistent to expect that it omits digits from `BigFloats`?

**Comment by StefanKarpinski** [comment]
We already omit trailing digits from long Float64 values, etc.:
```jl
julia> sqrt(2)
1.4142135623730951

julia> show(IOContext(STDOUT, :compact => true), sqrt(2))
1.41421
```
So I would expect that BigFloats would be truncated as well in compact mode. The whole point is to give some useful information while allowing a reasonable amount of data to be fit on the screen.

**Comment by jw3126** [comment]
You are right!

**Comment by fredrikekre** [comment]
Fixed by  #23388:
```julia
julia> show(IOContext(STDOUT, :compact => true), big(pi))
3.14159
```
